### ISSUE TITLE
Translation Memory problem

### ISSUE THREAD

**Issue by NightWalker77V** [URUWorks/TeroSubtitler#279]
Hello, and thanks for the improvements of the new version. But, I've noticed that the problem with the TM is still remains the same, which sadly makes it useless. So I've taken a step by step screenshots of what _I'm_ doing for the process and sending it in hopes that there's something I'm doing wrong and You could correct me. 
So here's what I do when I'm using the TM for the "first time" ( I don't have a .tmx file and want to create it from scratch)
1: open a source subtitle, then go presumably create a TM file from here:

![image](https://github.com/URUWorks/TeroSubtitler/assets/35186406/113001e1-7769-459d-b035-3944ccf6fa85)

2: Select the S and T languages and a name for the TM (I've tried it with/out the .tmx format) and then apply it :

![image](https://github.com/URUWorks/TeroSubtitler/assets/35186406/d21cb332-3fdc-4239-9438-eeb0d69064ba)

3: in here since I didn't add anything to the TM before for the reference, I'm opening a translated file to add it to the database and validate the translations for the source sentences:

![image](https://github.com/URUWorks/TeroSubtitler/assets/35186406/f8443892-e64e-41d4-bec0-9e4a800c35c5)

4 - After doing so, I expect that there's some data in the TM file, Which I don't know where is saved, and save and close the subtitle to open another one see what it's look like when I want to translate a new one with the help of TM. But as you see it just resets the whole process and there's no TM file that I can recall it to use it here, not in the subtitle folder not in the TM folder of the program it's nowhere! I have to do the process from No.2 again and it's frustrating... I even download a TM file and used it in the No.2 process ( hit the ... bottom in front of the name section) to see if I add the data it would work or not, But it still the same, there's nothing to recall for the next times:

![image](https://github.com/URUWorks/TeroSubtitler/assets/35186406/80a63550-3b73-4c4c-bfe0-20ec4cf799bb)

So could you please direct me in the right direction, If I'm doing it wrong? or if not. just fix the issue please. It's a key feature and I'm desperately want to be able to use it, Thanks in advance!

**Comment by URUWorks** [comment]
Yes, I need to correct several things, for example not specifying the memory folder, this should be saved in the "TM" subfolder but it is not, it may be saved in the Tero executable folder.
When reopening TM settings, there is a bug that does not update the file name...

Apart from the missing fixes, do you have suggestions to improve the use or its steps?

**Comment by NightWalker77V** [comment]
Surely I don't know much about the technical side of it, But for now, what's ideal are these things:

1- When opening a subtitle and wanting to use TM, there should be 2 options for users. A: use the existing TM or B: create a new TM. This way we can build multi-TMs for various subjects that each will use at certain times.

2- It should have a feature for support segment alignment. So even when a part of a sentence is available, the user can use that.

3- The ability to have and save different translations and meanings for the same sentence or phrase. Because, based on the scene, situation and context one phrase may have a different meaning, So if we can choose one of them it'd be nice

4- If we can export the TMs it's great for using it in another PC for example, or just be able to copy and backup the TM files locally.

5- The feature for updating the TM automatically, like every 10/30 minutes during the translation Or when the translation is done the user can save it manually for the final edit, it can appear when the user wants to close the program and asks "Do You want to save/update the TM?" 

All in all, these are the things that most TMs like Trados and like that, that are for text translation have in common. But, in Subtitle I've never seen it before, so some things should be done differently here. So I could provide more ideas when I use it in action
